### ISSUE TITLE
Develop Practically Perfect Prog Calculation use-case

### ISSUE THREAD

**Issue by TaraJensen** [dtcenter/METplus#406]
*Replace italics below with details for this issue.*

## Describe the New Feature ##
Develop a use case that computes the Practically Perfect Prog analysis.  Steps include:

1. Write a Python script to read in Local Storm Reports
2. Call ASCII2NC using Python Embedding to convert LSRs to MET netcdf - suggest script denotes 1,2,3 in QC area for Tornado, Hail, Wind reports respectively.
[Example of how to do this for ASCII2NC](https://ncar.github.io/METplus/Users_Guide/met_tool_wrapper/ASCII2NC/ASCII2NC_python_embedding.html#sphx-glr-users-guide-met-tool-wrapper-ascii2nc-ascii2nc-python-embedding-py)
3. Call Point2Grid to grid the LSRs
4. Use the binary output of Point2Grid tool as a mask.
5. Run RegridDataPlane with the MAXGAUSS method and a -width of 1 and -gaussian_radius of 120km to calculate practically perfect probabilities
6. Run Plot-Data-Plane

### Acceptance Testing ###
Input data:  SPC Local Storm Reports, csv files and kml files available - start with csv probably.  I believe a sample is available from Burkely - Lindsay to fill in where it is
Source:  https://www.spc.noaa.gov/climo/online/
Source: dakota:/d3/projects/MET/test_surrogate_severe/from_burkley_20200310/202005_rpts_filtered.csv
Test: Reproduce Practically Perfect Prog associated with the test case from Burkely

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*
*Larger issues should be broken down into sub-issues.*

### Relevant Deadlines ###
Needed for HWT, so 4/20/2020 would be best

### Funding Source ###
164951 - JTTI Project, PI: Romine  (Hank)
2770041 - JTTI Hail Project  (Lindsay and Tina)

### Assignee ###
- [ x] Select **engineer(s)** Hank Fisher
- [x ] Select **scientist(s)** Lindsay Blank and Tina Kalb

### Labels ###
- [x ] Select **component(s)**
- [ x] Select **priority**
- [x ] Select **requestor(s)**

### Projects and Milestone ###
- [x ] Review **projects** and select relevant **Repository** and **Organization** ones
- [x ] Select **milestone**

## Define Related Issue(s) ##
- [x ] Define new issue for METplus.
Related to [METplus Issue 405](https://github.com/NCAR/METplus/issues/405)
Related to [METplus Issue 414](https://github.com/NCAR/METplus/issues/414)
Related to [METplus Issue 414](https://github.com/NCAR/METplus/issues/414)

**Comment by CPKalb** [comment]
Should this be separate from the surrogate severe use case or a part of it?

**Comment by TaraJensen** [comment]
Not for now.  They are very different workflows. We need to add a separate issue to develop that use-case.

**Comment by lindsayrblank** [comment]
Burkely Gallo's practically perfect Fortran code
[practically_perfectnc.f90.txt](https://github.com/NCAR/METplus/files/4390409/practically_perfectnc.f90.txt)

**Comment by hankenstein2** [comment]
We have a python script used via python embedding to read the storm reports, we have a Point2Grid wrapper and are waiting an ascii2nc bug fix to chain the first 4 steps together.

**Comment by lindsayrblank** [comment]
Hi all, the example practically perfect data from Burkely Gallo can be found on dakota here:
dakota:/d3/projects/MET/test_surrogate_severe/from_burkley_20200310/
- Local Storm Reports CSV: 200205_rpts_filtered.csv
- Binary data of LSRs gridded to the NCEP 211 grid: 20200205_REPS211.dat
- LSRs gridded to the NCEP 211 grid: 20200205_REPS211.nc
- Practically perfect probabilities (finished product): 20200205_REPS211_PP.nc

**Comment by lindsayrblank** [comment]
Earlier today, we talked about threshold needed for practically perfect. I was incorrect when I said we used thresholds. The thresholds used in the surrogate severe calculations (issue #413 ) are thresholds of updraft helicity, which we don't use for practically perfect. Instead, the binary mask we need to calculate the practically perfect probability can be produced by the Point2Grid tool directly. Once we produce the binary mask, we then need to run the RegridDataPlane tool on it to create the probabilities. I updated the steps in the issue description to reflect the correct procedure.
